---
title: "RNA analysis"
author: "Matthew Beaumont"
date: "`r Sys.Date()`"
output: github_document
---

```{bash, eval=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

We obtained the following mRNA reads: 

```{bash}
cd /Volumes/Data/Projects/dmelR2_p-ele/rna/run1/raw
ls -lh *.fastq.gz

cd /Volumes/Data/Projects/dmelR2_p-ele/rna/run2/raw
ls -lh *.fastq.gz

```
We then ran fastQC on all files to assess their quality.

``` {bash eval=FALSE}
fastqc --outdir /Volumes/Data/Projects/dmelR2_p-ele/rna/raw/run1/fastQC /Volumes/Data/Projects/dmelR2_p-ele/rna/run1/raw/*fastq.gz
fastqc --outdir /Volumes/Data/Projects/dmelR2_p-ele/rna/raw/run2/fastQC /Volumes/Data/Projects/dmelR2_p-ele/rna/run2/raw/*fastq.gz

```

``` {bash}
cd /Volumes/Data/Projects/dmelR2_p-ele/rna/run1/fastQC/
ls -lh

cd /Volumes/Data/Projects/dmelR2_p-ele/rna/run2/fastQC/
ls -lh 

```

After this, we then realised that there was significant proportion of the 3' read ends (~10-30%) which were adapter sequences. To alleviate this, we decided to trim all of the 150bp reads down to 100bp, using the following for loop. 

``` {bash eval=FALSE}
for file in *gz; do
    gzip -cd "$file" | cut -c-100 | awk 'NR%4==2 && length($0)>=100{print prev; print $0; getline; print $0; getline; print $0} {prev=$0}' | gzip -c > "trimmed/${file}"
done

```

# Mapping

First, we indexed the D. mel + TE fasta file using samtools.

``` {bash eval=FALSE}
samtools faidx dmel_1215.4_rel_6_ISO1_consensusTEs.fasta

```

We then obtained the reference D. mel transcriptome and merged it with the list of consensus D. mel TEs, then indexed it.

``` {bash}
cd /Volumes/Data/Tools/RefGenomes/dmel/rna/dmel_TEs
ls -lh

```

Then we used bwa to map the forward and reverse reads to the reference.

``` {bash eval=FALSE}
nohup zsh dmel_RNA_mapping_bwamem.sh > ../logs/dmel_rna_map.log

```

dmel_RNA_mapping_bwamem.sh - 

``` {bash eval=FALSE}
ref="/Volumes/Data/Tools/RefGenomes/dmel/rna/dmel_TEs/dmel_1215.4_rel_6_ISO1_consensusTEs.fasta"
if="/Volumes/Data/Projects/dmelR2_p-ele/rna/raw/run2/trimmed"
of="/Volumes/Data/Projects/dmelR2_p-ele/rna/raw/run2/map-bwamem"

bwa mem -t 12 $ref $if/dmel_rna_R1_G6_run2_R1.fastq.gz $if/dmel_rna_R1_G6_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R1G6_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R2_G6_run2_R1.fastq.gz $if/dmel_rna_R2_G6_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R2G6_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R3_G6_run2_R1.fastq.gz $if/dmel_rna_R3_G6_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R3G6_run2.sort.bam

bwa mem -t 12 $ref $if/dmel_rna_R1_G15_run2_R1.fastq.gz $if/dmel_rna_R1_G15_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R1G15_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R2_G15_run2_R1.fastq.gz $if/dmel_rna_R2_G15_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R2G15_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R3_G15_run2_R1.fastq.gz $if/dmel_rna_R3_G15_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R3G15_run2.sort.bam

bwa mem -t 12 $ref $if/dmel_rna_R1_G21_run2_R1.fastq.gz $if/dmel_rna_R1_G21_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R1G21_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R2_G21_run2_R1.fastq.gz $if/dmel_rna_R2_G21_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R2G21_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R3_G21_run2_R1.fastq.gz $if/dmel_rna_R3_G21_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R3G21_run2.sort.bam

bwa mem -t 12 $ref $if/dmel_rna_R1_G30_run2_R1.fastq.gz $if/dmel_rna_R1_G30_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R1G30_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R2_G30_run2_R1.fastq.gz $if/dmel_rna_R2_G30_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R2G30_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R3_G30_run2_R1.fastq.gz $if/dmel_rna_R3_G30_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R3G30_run2.sort.bam

bwa mem -t 12 $ref $if/dmel_rna_R1_G40_run2_R1.fastq.gz $if/dmel_rna_R1_G40_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R1G40_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R2_G40_run2_R1.fastq.gz $if/dmel_rna_R2_G40_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R2G40_run2.sort.bam
bwa mem -t 12 $ref $if/dmel_rna_R3_G40_run2_R1.fastq.gz $if/dmel_rna_R3_G40_run2_R2.fastq.gz | samtools sort -m 2G --output-fmt BAM --threads 2 -o $of/dmel_R3G40_run2.sort.bam

```

# GSNAP

Run on Vetgrid06

To map via GMAP, we first need to create the database with the reference genome.

Then we need to build the database

``` {bash eval=FALSE}
gmap_build -D . -d mel-transcriptome /Volumes/Temp2/Matt/rna/refgenomes/dmel/dmel_1215.4_rel_6_ISO1_consensusTEs.fasta

```

Then we run the following script to for alignment, while assessing for novel spicing events.

``` {bash eval=FALSE}
#!/bin/bash

refdir="/Volumes/Temp2/Matt/rna/refgenomes/GMAP/mel_database"
refname="mel_database"

input_dir="/Volumes/Temp2/Matt/rna/trimmed/trimmed"
output_dir="/Volumes/Temp2/Matt/rna/map_GSNAP/output"

gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R1_G6_run2_R1.fastq.gz" "$input_dir/dmel_rna_R1_G6_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R1G6.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R2_G6_run2_R1.fastq.gz" "$input_dir/dmel_rna_R2_G6_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R2G6.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R3_G6_run2_R1.fastq.gz" "$input_dir/dmel_rna_R3_G6_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R3G6.sort.bam"

gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R1_G15_run2_R1.fastq.gz" "$input_dir/dmel_rna_R1_G15_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R1G15.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R2_G15_run2_R1.fastq.gz" "$input_dir/dmel_rna_R2_G15_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R2G15.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R3_G15_run2_R1.fastq.gz" "$input_dir/dmel_rna_R3_G15_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R3G15.sort.bam"

gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R1_G21_run2_R1.fastq.gz" "$input_dir/dmel_rna_R1_G21_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R1G21.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R2_G21_run2_R1.fastq.gz" "$input_dir/dmel_rna_R2_G21_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R2G21.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R3_G21_run2_R1.fastq.gz" "$input_dir/dmel_rna_R3_G21_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R3G21.sort.bam"

gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R1_G30_run2_R1.fastq.gz" "$input_dir/dmel_rna_R1_G30_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R1G30.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R2_G30_run2_R1.fastq.gz" "$input_dir/dmel_rna_R2_G30_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R2G30.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R3_G30_run2_R1.fastq.gz" "$input_dir/dmel_rna_R3_G30_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R3G30.sort.bam"

gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R1_G40_run2_R1.fastq.gz" "$input_dir/dmel_rna_R1_G40_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R1G40.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R2_G40_run2_R1.fastq.gz" "$input_dir/dmel_rna_R2_G40_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R2G40.sort.bam"
gsnap -D "$refdir" -d "$refname" -t 12 -N 1 --format=sam --gunzip "$input_dir/dmel_rna_R3_G40_run2_R1.fastq.gz" "$input_dir/dmel_rna_R3_G40_run2_R2.fastq.gz" | samtools sort -m 2G --output-fmt BAM --threads 2 -o "$output_dir/gt_R3G40.sort.bam"

```

Which provides the following sorted BAM files.


# Coverage 

We ran the following script to assess coverage.

``` {bash eval=FALSE}
nohup zsh expression-splicing.sh > ../logs/splicing-expression.log

```

``` {bash eval=FALSE}
#!/bin/bash

fai="/Volumes/Data/Tools/RefGenomes/dmel/rna/dmel_TEs/dmel_1215.4_rel_6_ISO1_consensusTEs.fasta.fai"
pyscript="/Volumes/Data/Projects/dmelR2_p-ele/scripts/mRNA-coverage-senseantisense.py"
input_dir="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/map-bwamem"
output_dir="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/raw_expression"
seqs="PPI251,FBtr0141238_mRNA,FBtr0141090_mRNA,FBtr0132023_mRNA,FBtr0137012_mRNA,FBtr0140282_mRNA,FBtr0141095_mRNA,FBtr0142660_mRNA,FBtr0140302_mRNA,FBtr0142342_mRNA,FBtr0143902_mRNA,FBtr0130551_mRNA,FBtr0145198_mRNA,FBtr0130342_mRNA,FBtr0135961_mRNA,FBtr0135197_mRNA,FBtr0130390_mRNA,FBtr0141576_mRNA,FBtr0139614_mRNA,FBtr0130391_mRNA,FBtr0143034_mRNA"

samtools view $input_dir/dmel_R1G6_run2.sort.bam | python $pyscript --sam - --sample-id R1-G6-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R1G6_run2.txt
samtools view $input_dir/dmel_R2G6_run2.sort.bam | python $pyscript --sam - --sample-id R2-G6-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R2G6_run2.txt
samtools view $input_dir/dmel_R3G6_run2.sort.bam | python $pyscript --sam - --sample-id R3-G6-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R3G6_run2.txt

samtools view $input_dir/dmel_R1G15_run2.sort.bam | python $pyscript --sam - --sample-id R1-G15-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R1G15_run2.txt
samtools view $input_dir/dmel_R2G15_run2.sort.bam | python $pyscript --sam - --sample-id R2-G15-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R2G15_run2.txt
samtools view $input_dir/dmel_R3G15_run2.sort.bam | python $pyscript --sam - --sample-id R3-G15-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R3G15_run2.txt

samtools view $input_dir/dmel_R1G21_run2.sort.bam | python $pyscript --sam - --sample-id R1-G21-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R1G21_run2.txt
samtools view $input_dir/dmel_R2G21_run2.sort.bam | python $pyscript --sam - --sample-id R2-G21-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R2G21_run2.txt
samtools view $input_dir/dmel_R3G21_run2.sort.bam | python $pyscript --sam - --sample-id R3-G21-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R3G21_run2.txt

samtools view $input_dir/dmel_R1G30_run2.sort.bam | python $pyscript --sam - --sample-id R1-G30-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R1G30_run2.txt
samtools view $input_dir/dmel_R2G30_run2.sort.bam | python $pyscript --sam - --sample-id R2-G30-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R2G30_run2.txt
samtools view $input_dir/dmel_R3G30_run2.sort.bam | python $pyscript --sam - --sample-id R3-G30-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R3G30_run2.txt

samtools view $input_dir/dmel_R1G40_run2.sort.bam | python $pyscript --sam - --sample-id R1-G40-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R1G40_run2.txt
samtools view $input_dir/dmel_R2G40_run2.sort.bam | python $pyscript --sam - --sample-id R2-G40-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R2G40_run2.txt
samtools view $input_dir/dmel_R3G40_run2.sort.bam | python $pyscript --sam - --sample-id R3-G40-run2 --seqs $seqs --fai $fai > $output_dir/dmel_R3G40_run2.txt

cat *.txt| perl -pe 's/-/\t/'|perl -pe 's/-/\t/' > expr-spli.forr

```

# P-element expression

We ran the following script to extract out all expression levels.

``` {bash eval=FALSE}
nohup zsh /Volumes/Data/Projects/dmelR2_p-ele/scripts/dmel_pele_expression.sh > ../../../../../logs/pele-expression.log

```

``` {bash eval=FALSE}
#!/bin/bash

fai="/Volumes/Data/Tools/RefGenomes/dmel/rna/dmel_TEs/dmel_1215.4_rel_6_ISO1_consensusTEs.fasta.fai"
pyscript="/Volumes/Data/Projects/dmelR2_p-ele/scripts/mRNA-expression.py"
input_dir="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/map-bwamem"
output_dir="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/raw_expression/all-expressionlevel"

samtools view $input_dir/dmel_R1G6_run2.sort.bam | python $pyscript --sam - --sample-id R1-G6-run2 --fai $fai > $output_dir/expr_dmel_R1G6_run2.txt
samtools view $input_dir/dmel_R2G6_run2.sort.bam | python $pyscript --sam - --sample-id R2-G6-run2 --fai $fai > $output_dir/expr_dmel_R2G6_run2.txt
samtools view $input_dir/dmel_R3G6_run2.sort.bam | python $pyscript --sam - --sample-id R3-G6-run2 --fai $fai > $output_dir/expr_dmel_R3G6_run2.txt

samtools view $input_dir/dmel_R1G15_run2.sort.bam | python $pyscript --sam - --sample-id R1-G15-run2 --fai $fai > $output_dir/expr_dmel_R1G15_run2.txt
samtools view $input_dir/dmel_R2G15_run2.sort.bam | python $pyscript --sam - --sample-id R2-G15-run2 --fai $fai > $output_dir/expr_dmel_R2G15_run2.txt
samtools view $input_dir/dmel_R3G15_run2.sort.bam | python $pyscript --sam - --sample-id R3-G15-run2 --fai $fai > $output_dir/expr_dmel_R3G15_run2.txt

samtools view $input_dir/dmel_R1G21_run2.sort.bam | python $pyscript --sam - --sample-id R1-G21-run2 --fai $fai > $output_dir/expr_dmel_R1G21_run2.txt
samtools view $input_dir/dmel_R2G21_run2.sort.bam | python $pyscript --sam - --sample-id R2-G21-run2 --fai $fai > $output_dir/expr_dmel_R2G21_run2.txt
samtools view $input_dir/dmel_R3G21_run2.sort.bam | python $pyscript --sam - --sample-id R3-G21-run2 --fai $fai > $output_dir/expr_dmel_R3G21_run2.txt

samtools view $input_dir/dmel_R1G30_run2.sort.bam | python $pyscript --sam - --sample-id R1-G30-run2 --fai $fai > $output_dir/expr_dmel_R1G30_run2.txt
samtools view $input_dir/dmel_R2G30_run2.sort.bam | python $pyscript --sam - --sample-id R2-G30-run2 --fai $fai > $output_dir/expr_dmel_R2G30_run2.txt
samtools view $input_dir/dmel_R3G30_run2.sort.bam | python $pyscript --sam - --sample-id R3-G30-run2 --fai $fai > $output_dir/expr_dmel_R3G30_run2.txt

samtools view $input_dir/dmel_R1G40_run2.sort.bam | python $pyscript --sam - --sample-id R1-G40-run2 --fai $fai > $output_dir/expr_dmel_R1G40_run2.txt
samtools view $input_dir/dmel_R2G40_run2.sort.bam | python $pyscript --sam - --sample-id R2-G40-run2 --fai $fai > $output_dir/expr_dmel_R2G40_run2.txt
samtools view $input_dir/dmel_R3G40_run2.sort.bam | python $pyscript --sam - --sample-id R3-G40-run2 --fai $fai > $output_dir/expr_dmel_R3G40_run2.txt

cat *.txt| perl -pe 's/-/\t/'|perl -pe 's/-/\t/' > expr.forr

```

Then we visualise it using ggplot.

```{R}
library(tidyverse)
library(RColorBrewer)
theme_set(theme_bw())
tresrep<-c("#e41a1c","#4daf4a","#377eb8")

t=read_delim("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/raw_expression/all-expressionlevel/expr.forr",delim="\t",col_names=FALSE,comment="#")
names(t)<-c("replicate","generation","run","gene","rawse","rawase","genlen","sense","antisense","total")
t<-subset(t,gene=="PPI251")
t$generation<-as.numeric(substring(t$generation, 2))
width <- 16
height <- 12
resolution <- 600

s<-ggplot()+geom_line(data=t,aes(x=generation,y=sense,color=replicate),linewidth=1)+
  geom_point(data=t,aes(x=generation,y=sense,color=replicate))+
  theme(strip.text=element_blank(),legend.position=c(0.1,0.9))+
  ylab("expression [rpkm]")+scale_colour_manual(values=tresrep)+xlim(0,48)

plot(s)

ggsave("P-ele_expression.png", plot = s, width = 16, height = 12, dpi = 600)

```

# mRNA overview 

```{bash eval=FALSE}
pyscript="/Users/rokofler/dev/te-tools/ere/expression/mRNA-overview.py"
if="/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/map-gsnap-trans"
of="/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/05-overview"

samtools view $if/gt_wf_R1G5.sort.bam | python $pyscript --sam - --sample-id R1-G5-wf   > $of/wf_R1G5.txt
samtools view $if/gt_wf_R2G5.sort.bam | python $pyscript --sam - --sample-id R2-G5-wf   > $of/wf_R2G5.txt
samtools view $if/gt_wf_R4G5.sort.bam | python $pyscript --sam - --sample-id R4-G5-wf   > $of/wf_R4G5.txt

samtools view $if/gt_wf_R1G15.sort.bam | python $pyscript --sam - --sample-id R1-G15-wf   > $of/wf_R1G15.txt
samtools view $if/gt_wf_R2G15.sort.bam | python $pyscript --sam - --sample-id R2-G15-wf   > $of/wf_R2G15.txt
samtools view $if/gt_wf_R4G15.sort.bam | python $pyscript --sam - --sample-id R4-G15-wf   > $of/wf_R4G15.txt

samtools view $if/gt_wf_R1G20.sort.bam | python $pyscript --sam - --sample-id R1-G20-wf   > $of/wf_R1G20.txt
samtools view $if/gt_wf_R2G20.sort.bam | python $pyscript --sam - --sample-id R2-G20-wf   > $of/wf_R2G20.txt
samtools view $if/gt_wf_R4G20.sort.bam | python $pyscript --sam - --sample-id R4-G20-wf   > $of/wf_R4G20.txt

samtools view $if/gt_wf_R1G30.sort.bam | python $pyscript --sam - --sample-id R1-G30-wf   > $of/wf_R1G30.txt
samtools view $if/gt_wf_R2G30.sort.bam | python $pyscript --sam - --sample-id R2-G30-wf   > $of/wf_R2G30.txt
samtools view $if/gt_wf_R4G30.sort.bam | python $pyscript --sam - --sample-id R4-G30-wf   > $of/wf_R4G30.txt

samtools view $if/gt_wf_R1G40.sort.bam | python $pyscript --sam - --sample-id R1-G40-wf   > $of/wf_R1G40.txt
samtools view $if/gt_wf_R2G40.sort.bam | python $pyscript --sam - --sample-id R2-G40-wf   > $of/wf_R2G40.txt
samtools view $if/gt_wf_R4G40.sort.bam | python $pyscript --sam - --sample-id R4-G40-wf   > $of/wf_R4G40.txt

cat *txt|perl -pe 's/-/\t/g'

```

```{R eval=FALSE}
library(ggplot2)
theme_set(theme_bw())

genes<-c("PPI251","FBtr0137012_mRNA","FBtr0140282_mRNA","FBtr0141095_mRNA","FBtr0142660_mRNA","FBtr0140302_mRNA",
         "FBtr0142342_mRNA","FBtr0143902_mRNA","FBtr0130551_mRNA","FBtr0145198_mRNA","FBtr0130342_mRNA","FBtr0135961_mRNA",
         "FBtr0135197_mRNA","FBtr0130390_mRNA","FBtr0141576_mRNA","FBtr0139614_mRNA","FBtr0130391_mRNA",
         "FBtr0141238_mRNA","FBtr0141090_mRNA","FBtr0132023_mRNA","FBtr0143034_mRNA","FBtr0141271_mRNA")
sc<-c("PPI251","spnE","cuff","Dcr2","Hen1","Hen1","krimp","loqs","r2d2","vas",
      "zuc","ago2","armi","aub","del","moon","piwi","tj","rhino","rpl32","qin","Chk2")


h<-read.table("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/mRNA/overview/mRNA_all_forr.txt")        
names(h)<-c("rep","time","tissue","strand","gene","pos","cov")

for(i in 1:length(genes))
{
  target<-genes[i]
  short<-sc[i]
  
  a<-subset(h,gene==target)
  a$time <- factor(a$time, levels=c("G6", "G15", "G21", "G30", "G40"))
  
  s<-subset(a,strand=="se")
  as<-subset(a,strand=="ase")
  
  
  plot <- ggplot() +
    geom_polygon(data=s,mapping=aes(x=pos, y=cov), fill='grey', color='grey') +
    geom_polygon(data=as, aes(x=pos, y=-cov), fill='lightgrey', color='lightgrey')+
    facet_grid(time~rep)

  plot(plot)
}
```


# Chk2 - Wilcoxon Test

```{R}
library(ggplot2)
library(tidyverse)
theme_set(theme_bw())

h<-read.table("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/mRNA/overview/mRNA_all_forr.txt")        

names(h)<-c("rep","time","tissue","strand","gene","pos","cov")
a<-subset(h,gene=="FBtr0141271_mRNA" & strand=="se")
mc<-a %>% group_by(rep,time) %>% summarize(mcov=mean(cov))
wilcox.test(subset(mc,rep=="R2")$mcov,subset(mc,rep!="R2")$mcov)
```

# Splicing & expression visualisation

``` {bash eval=FALSE}
#!/bin/bash

fai="/Volumes/Data/Tools/RefGenomes/dmel/rna/dmel_TEs/dmel_1215.4_rel_6_ISO1_consensusTEs.fasta.fai"
pyscript="/Volumes/Data/Projects/dmelR2_p-ele/scripts/mRNA-coverage-senseantisense.py"
if="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/map-GMAP/output"
of="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/expression"
seqs="PPI251,FBtr0141238_mRNA,FBtr0138851_mRNA,FBtr0141090_mRNA,FBtr0132023_mRNA,FBtr0137012_mRNA,FBtr0140282_mRNA,FBtr0141095_mRNA,FBtr0142660_mRNA,FBtr0140302_mRNA,FBtr0142342_mRNA,FBtr0143902_mRNA,FBtr0130551_mRNA,FBtr0145198_mRNA,FBtr0130342_mRNA,FBtr0135961_mRNA,FBtr0135197_mRNA,FBtr0130390_mRNA,FBtr0141576_mRNA,FBtr0139614_mRNA,FBtr0130391_mRNA,FBtr0143034_mRNA"

samtools view $if/gt_R1G6.sort.bam | python $pyscript --sam - --sample-id R1-G6 --seqs $seqs --fai $fai -a  > $of/R1G6.txt
samtools view $if/gt_R2G6.sort.bam | python $pyscript --sam - --sample-id R2-G6 --seqs $seqs --fai $fai -a  > $of/R2G6.txt
samtools view $if/gt_R3G6.sort.bam | python $pyscript --sam - --sample-id R3-G6 --seqs $seqs --fai $fai -a  > $of/R3G6.txt

samtools view $if/gt_R1G15.sort.bam | python $pyscript --sam - --sample-id R1-G15 --seqs $seqs --fai $fai -a  > $of/R1G15.txt
samtools view $if/gt_R2G15.sort.bam | python $pyscript --sam - --sample-id R2-G15 --seqs $seqs --fai $fai -a  > $of/R2G15.txt
samtools view $if/gt_R3G15.sort.bam | python $pyscript --sam - --sample-id R3-G15 --seqs $seqs --fai $fai -a  > $of/R3G15.txt

samtools view $if/gt_R1G21.sort.bam | python $pyscript --sam - --sample-id R1-G21 --seqs $seqs --fai $fai -a  > $of/R1G21.txt
samtools view $if/gt_R2G21.sort.bam | python $pyscript --sam - --sample-id R2-G21 --seqs $seqs --fai $fai -a  > $of/R2G21.txt
samtools view $if/gt_R3G21.sort.bam | python $pyscript --sam - --sample-id R3-G21 --seqs $seqs --fai $fai -a  > $of/R3G21.txt

samtools view $if/gt_R1G30.sort.bam | python $pyscript --sam - --sample-id R1-G30 --seqs $seqs --fai $fai -a  > $of/R1G30.txt
samtools view $if/gt_R2G30.sort.bam | python $pyscript --sam - --sample-id R2-G30 --seqs $seqs --fai $fai -a  > $of/R2G30.txt
samtools view $if/gt_R3G30.sort.bam | python $pyscript --sam - --sample-id R3-G30 --seqs $seqs --fai $fai -a  > $of/R3G30.txt

samtools view $if/gt_R1G40.sort.bam | python $pyscript --sam - --sample-id R1-G40 --seqs $seqs --fai $fai -a  > $of/R1G40.txt
samtools view $if/gt_R2G40.sort.bam | python $pyscript --sam - --sample-id R2-G40 --seqs $seqs --fai $fai -a  > $of/R2G40.txt
samtools view $if/gt_R3G40.sort.bam | python $pyscript --sam - --sample-id R3-G40 --seqs $seqs --fai $fai -a  > $of/R3G40.txt

# Combine all files together in the correct format afterwards.
#cat *.txt| perl -pe 's/-/\t/'|perl -pe 's/-/\t/' > expr.forr 

#awk 'BEGIN{OFS="\t"}{$2 = $2 "\t" "wf"; print}' expr.forr > expr_wf.forr

```

Then for splicing.

```{bash eval=FALSE}
#!/bin/bash

pyscript="/Volumes/Data/Projects/dmelR2_p-ele/scripts/mRNA-splicing-senseantisense.py"
if="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/map-GMAP/output"
of="/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/splicing"
seqs="PPI251,FBtr0141238_mRNA,FBtr0138851_mRNA,FBtr0141090_mRNA,FBtr0132023_mRNA,FBtr0137012_mRNA,FBtr0140282_mRNA,FBtr0141095_mRNA,FBtr0142660_mRNA,FBtr0140302_mRNA,FBtr0142342_mRNA,FBtr0143902_mRNA,FBtr0130551_mRNA,FBtr0145198_mRNA,FBtr0130342_mRNA,FBtr0135961_mRNA,FBtr0135197_mRNA,FBtr0130390_mRNA,FBtr0141576_mRNA,FBtr0139614_mRNA,FBtr0130391_mRNA,FBtr0143034_mRNA"

samtools view $if/gt_R1G6.sort.bam | python $pyscript --sam - --sample-id R1-G6-wf --seqs $seqs  > $of/R1G6.txt
samtools view $if/gt_R2G6.sort.bam | python $pyscript --sam - --sample-id R2-G6-wf --seqs $seqs  > $of/R2G6.txt
samtools view $if/gt_R3G6.sort.bam | python $pyscript --sam - --sample-id R3-G6-wf --seqs $seqs  > $of/R3G6.txt

samtools view $if/gt_R1G15.sort.bam | python $pyscript --sam - --sample-id R1-G15-wf --seqs $seqs  > $of/R1G15.txt
samtools view $if/gt_R2G15.sort.bam | python $pyscript --sam - --sample-id R2-G15-wf --seqs $seqs  > $of/R2G15.txt
samtools view $if/gt_R3G15.sort.bam | python $pyscript --sam - --sample-id R3-G15-wf --seqs $seqs  > $of/R3G15.txt

samtools view $if/gt_R1G21.sort.bam | python $pyscript --sam - --sample-id R1-G21-wf --seqs $seqs  > $of/R1G21.txt
samtools view $if/gt_R2G21.sort.bam | python $pyscript --sam - --sample-id R2-G21-wf --seqs $seqs  > $of/R2G21.txt
samtools view $if/gt_R3G21.sort.bam | python $pyscript --sam - --sample-id R3-G21-wf --seqs $seqs  > $of/R3G21.txt

samtools view $if/gt_R1G30.sort.bam | python $pyscript --sam - --sample-id R1-G30-wf --seqs $seqs  > $of/R1G30.txt
samtools view $if/gt_R2G30.sort.bam | python $pyscript --sam - --sample-id R2-G30-wf --seqs $seqs  > $of/R2G30.txt
samtools view $if/gt_R3G30.sort.bam | python $pyscript --sam - --sample-id R3-G30-wf --seqs $seqs  > $of/R3G30.txt

samtools view $if/gt_R1G40.sort.bam | python $pyscript --sam - --sample-id R1-G40-wf --seqs $seqs  > $of/R1G40.txt
samtools view $if/gt_R2G40.sort.bam | python $pyscript --sam - --sample-id R2-G40-wf --seqs $seqs  > $of/R2G40.txt
samtools view $if/gt_R3G40.sort.bam | python $pyscript --sam - --sample-id R3-G40-wf --seqs $seqs  > $of/R3G40.txt

# Combine all .txt files win the correct format, splitting Sample-ID.
# cat *.txt| perl -pe 's/-/\t/'|perl -pe 's/-/\t/' > spli.forr

```

```{R}
library(ggplot2)
theme_set(theme_bw())


#genes<-c("PPI251","FBtr0137012_mRNA","FBtr0140282_mRNA","FBtr0141095_mRNA","FBtr0142660_mRNA","FBtr0140302_mRNA",
#         "FBtr0142342_mRNA","FBtr0143902_mRNA","FBtr0130551_mRNA","FBtr0145198_mRNA","FBtr0130342_mRNA","FBtr0135961_mRNA",
#         "FBtr0135197_mRNA","FBtr0130390_mRNA","FBtr0141576_mRNA","FBtr0139614_mRNA","FBtr0130391_mRNA",
#         "FBtr0141238_mRNA","FBtr0141090_mRNA","FBtr0132023_mRNA","FBtr0143034_mRNA")
#sc<-c("PPI251","spnE","cuff","Dcr2","Hen1","Hen1","krimp","loqs","r2d2","vas",

target<-"PPI251"
ttissue<-"wf" # target tissue
sminfreq<-0.1

h<-read.table("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/expression/expr_wf.forr")
names(h)<-c("rep","time","tissue","strand","gene","pos","cov")

a<-subset(h,gene==target & tissue==ttissue)
a$key<-paste0(a$rep,"_",a$time,"_",a$pos,"_",a$strand)


a$time <- factor(a$time, levels=c("G6", "G15", "G21", "G30", "G40"))
s<-subset(a,strand=="se")
as<-subset(a,strand=="ase")

spli<-read.table("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/splicing/spli.forr")

names(spli)<-c("rep","time","tissue","strand","gene","skey","start","end","rawcount","freq")
aspli<-subset(spli,gene==target & freq>sminfreq & tissue==ttissue)
aspli$time <- factor(aspli$time, levels=c("G6", "G15", "G21", "G30", "G40"))
aspli$rep<- as.factor(aspli$rep)
aspli$start<-aspli$start-1 # position inaccuracy, graph is more appealing
aspli$keystart<-paste0(aspli$rep,"_",aspli$time,"_",aspli$start,"_",aspli$strand)
aspli$keyend<-paste0(aspli$rep,"_",aspli$time,"_",aspli$end,"_",aspli$strand)


aspli<-merge(x=aspli,y=a[,c("key","cov")],by.x="keystart",by.y="key")
aspli<-merge(x=aspli,y=a[,c("key","cov")],by.x="keyend",by.y="key")
aspli$size<-log(aspli$freq+1)+1


a_s<-subset(aspli,strand=="se")
a_as<-subset(aspli,strand=="ase")
  
  
plot <- ggplot() +
    geom_polygon(data=s,mapping=aes(x=pos, y=cov), fill='grey', color='grey') +
    geom_polygon(data=as, aes(x=pos, y=-cov), fill='lightgrey', color='lightgrey')+
    geom_curve(data=a_s, mapping=aes(x=start, y=cov.x, xend=end, yend=cov.y,linewidth=size), curvature=-0.2, ncp=10, show.legend=FALSE)+
    geom_curve(data=a_as, mapping=aes(x=start, y=-cov.x, xend=end, yend=-cov.y,linewidth=size), curvature=0.2, ncp=10, show.legend=FALSE)+
    facet_grid(time~rep)+scale_size(range=c(0.2,2))+xlab("position")+ylab("expression level [rpm]")

#     q <- p + geom_curve(data=coords, mapping=aes(x=x, y=y, xend=xend, yend=yend, 
# size=size), curvature=-0.2, ncp=10, show.legend=FALSE) +
#  scale_size(range = sizerange)

plot(plot)
outfile<-"/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/PPI251_wf.pdf"
  
  pdf(file=outfile,width=8,height=8)
  plot(plot)
  #dev.off()
  
```
## IVS3 

```{R}
library(tidyverse)
library(RColorBrewer)
theme_set(theme_bw())
tresrep<-c("#e41a1c","#377eb8","#4daf4a")

# Annotation
# PPI251	ensembl	exon	153	442	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	501	1168	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	1222	1947	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	2138	2709	.	+	.	gene_id "pele"; transcript_id "pele1";

t<-read.table("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/splicing/spli.forr")
# head
# R1	G10	ov	se	FBtr0143902_mRNA	1009-1151	1009	1151	1	0.0220242525784
#R1	G10	ov	se	FBtr0143902_mRNA	1009-1148	1009	1148	1	0.0220242525784
names<-c("rep","time","tissue","strand","gene","skey","start","end","rawcount","freq")
names(t)<-names
t<-subset(t,gene=="PPI251" & tissue=="wf" & strand=="se")
# t[t$time=="naive",]$time<-"G0" # lets use the naive flies as generation 0; but no splicing in naive flies
t$time<-as.numeric(substring(t$time, 2))

t<-subset(t,skey=="1948-2139") # delta23
missingvalues=data.frame(rep=c("R1","R2","R3"),time=c(0,0,0),tissue=rep("wf",3),
                         strand=rep("se",3),gene=rep("PPI251",3),skey=rep("1169-1223",3),
                        start=rep(0,3),end=rep(0,3),rawcount=rep(0,3),freq=rep(0,3))
# G0, 124
# G5,  1
# G30, 4
# G40, 14
t<-rbind(t,missingvalues)
#t<-subset(t,skey=="443-502") # delta12


s<-ggplot()+geom_line(data=t,aes(x=time,y=freq,color=rep),size=1)+
  geom_point(data=t,aes(x=time,y=freq,color=rep))+
  theme(strip.text=element_blank(),legend.position=c(0.1,0.9))+

  ylab("splicing level IVS3 [srpm]")+scale_colour_manual(values=tresrep)+xlim(0,48)

plot(s)
#postscript(file="/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/04-visualize-exprAndSplice/graph/IVS3_expression.ps",width=5.95,height=2.5)
#plot(s)
#dev.off()
```

## Linegraph IVS2
```{R}
library(tidyverse)
library(RColorBrewer)
theme_set(theme_bw())
tresrep<-c("#e41a1c","#377eb8","#4daf4a")

# Annotation
# PPI251	ensembl	exon	153	442	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	501	1168	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	1222	1947	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	2138	2709	.	+	.	gene_id "pele"; transcript_id "pele1";

t<-read.table("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/splicing/spli.forr")
# head
# R1	G10	ov	se	FBtr0143902_mRNA	1009-1151	1009	1151	1	0.0220242525784
#R1	G10	ov	se	FBtr0143902_mRNA	1009-1148	1009	1148	1	0.0220242525784
names<-c("rep","time","tissue","strand","gene","skey","start","end","rawcount","freq")
names(t)<-names
t<-subset(t,gene=="PPI251" & tissue=="wf" & strand=="se")
# t[t$time=="naive",]$time<-"G0" # lets use the naive flies as generation 0; but no splicing in naive flies
t$time<-as.numeric(substring(t$time, 2))

t<-subset(t,skey=="1169-1223")
missingvalues=data.frame(rep=c("R1","R2","R3"),time=c(0,0,0),tissue=rep("wf",3),
                         strand=rep("se",3),gene=rep("PPI251",3),skey=rep("1169-1223",3),
                        start=rep(0,3),end=rep(0,3),rawcount=rep(0,3),freq=rep(0,3))
# G0, 124
# G5,  1
# G30, 4
# G40, 14
t<-rbind(t,missingvalues)



s<-ggplot()+geom_line(data=t,aes(x=time,y=freq,color=rep),size=1)+
  geom_point(data=t,aes(x=time,y=freq,color=rep))+
  theme(strip.text=element_blank(),legend.position=c(0.1,0.9))+
  #  geom_line(data=hot,aes(x=generation,y=u,color=replicate),linetype="dashed")+
  ylab("splicing level IVS2 [rpm]")+scale_colour_manual(values=tresrep)+xlim(0,48)

plot(s)
#pdf(file="/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/04-visualize-exprAndSplice/graph/IVS2_expression.pdf",width=5.95,height=2.5)
#plot(s)
#dev.off()
```

## Linegraph IVS1
```{R}
library(tidyverse)
library(RColorBrewer)
theme_set(theme_bw())
tresrep<-c("#e41a1c","#377eb8","#4daf4a")

# Annotation
# PPI251	ensembl	exon	153	442	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	501	1168	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	1222	1947	.	+	.	gene_id "pele"; transcript_id "pele1";
# PPI251	ensembl	exon	2138	2709	.	+	.	gene_id "pele"; transcript_id "pele1";

t<-read.table("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/splicing/spli.forr")
# head
# R1	G10	ov	se	FBtr0143902_mRNA	1009-1151	1009	1151	1	0.0220242525784
#R1	G10	ov	se	FBtr0143902_mRNA	1009-1148	1009	1148	1	0.0220242525784
names<-c("rep","time","tissue","strand","gene","skey","start","end","rawcount","freq")
names(t)<-names
t<-subset(t,gene=="PPI251" & tissue=="wf" & strand=="se")
# t[t$time=="naive",]$time<-"G0" # lets use the naive flies as generation 0; but no splicing in naive flies
t$time<-as.numeric(substring(t$time, 2))

t<-subset(t,skey=="443-502")
missingvalues=data.frame(rep=c("R1","R2","R3"),time=c(0,0,0),tissue=rep("wf",3),
                         strand=rep("se",3),gene=rep("PPI251",3),skey=rep("443-502",3),
                        start=rep(0,3),end=rep(0,3),rawcount=rep(0,3),freq=rep(0,3))
# G0, 124
# G5,  1
# G30, 4
# G40, 14
t<-rbind(t,missingvalues)



s<-ggplot()+geom_line(data=t,aes(x=time,y=freq,color=rep),size=1,lty=1)+
   geom_point(data=t,aes(x=time,y=freq,color=rep))+
  theme(strip.text=element_blank(),legend.position=c(0.1,0.9))+
  #  geom_line(data=hot,aes(x=generation,y=u,color=replicate),linetype="dashed")+
  ylab("splicing level IVS1 [rpm]")+scale_colour_manual(values=tresrep)+xlim(0,48)

plot(s)
#pdf(file="/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/04-visualize-exprAndSplice/graph/IVS1_expression.pdf",width=5.95,height=2.5)
#plot(s)
#dev.off()
```

# edge-R

Prepped for addition of naive datasets.

```{R eval=FALSE}
library(tidyverse)
library(RColorBrewer)
library(dplyr)
library(edgeR)
library(data.table)
library(gridExtra)
#library(gridExtra)
theme_set(theme_bw())
palette <- c("darkgrey","orange","red")
colour<-c("grey50","#e41a1c","#377eb8","#4daf4a","#984ea3","#ff7f00","#a65628","#f781bf")
cbbPalette<-colour

d<-read_delim("/Volumes/Data/Projects/dmelR2_p-ele/rna/run2/splicing-expression/raw_expression/all-expressionlevel/expr.forr",delim="\t",col_names=FALSE,comment="#")
names(d)<-c("replicate","generation","run","gene","rawse","rawase","genlen","se","ase","total")


dgeme <- function(cou,gro) {
  y <- DGEList(counts=cou,group=gro)
  keep <- filterByExpr(y)
  y <- y[keep,,keep.lib.sizes=FALSE]
  y <- calcNormFactors(y)
  design <- model.matrix(~gro)
  y <- estimateDisp(y,design)
  #fit <- glmFit(y,design)
  #lrt <- glmLRT(fit,coef=2)
  fit <- glmQLFit(y,design)
  lrt <- glmQLFTest(fit,coef=2)
  return(lrt)
  
}

edgeme <- function(n1,n2,n3,e1,e2) {
  merge<- n1 %>%  inner_join(n2, by = "gene")  %>%  inner_join(n3, by = "gene") %>%  inner_join(e1, by = "gene") %>%  inner_join(e2, by = "gene")
  s<-setDT(merge)
  dm<-data.matrix(s[,2:6])
  rownames(dm)<-merge$gene
  group <- factor(c(1,1,1,2,2))
  lrt<-dgeme(dm,group)
  return(lrt)
}

edgemeext <- function(n1,n2,n3,e5,e1,e2) {
  merge<- n1 %>%  inner_join(n2, by = "gene")  %>%  inner_join(n3, by = "gene") %>% inner_join(e5, by = "gene") %>%  inner_join(e1, by = "gene") %>%  inner_join(e2, by = "gene")
  s<-setDT(merge)
  dm<-data.matrix(s[,2:7])
  rownames(dm)<-merge$gene
  group <- factor(c(1,1,1,1,2,2))
  lrt<-dgeme(dm,group)
  return(lrt)
}

# Need naive data for diff expression.

naive<-subset(d,generation=="naive")
nR1<-subset(naive,replicate=="R1") %>% select(gene,rawse) %>% rename(naiveR1=rawse)
nR2<-subset(naive,replicate=="R2") %>% select(gene,rawse) %>% rename(naiveR2=rawse)
nR3<-subset(naive,replicate=="R3") %>% select(gene,rawse) %>% rename(naiveR3=rawse)


rep1<-subset(d,replicate=="R1")
eg6R1<-subset(rep1,generation=="G6") %>% select(gene,rawse) %>% rename(eg6R1=rawse)
eg15R1<-subset(rep1,generation=="G15") %>% select(gene,rawse) %>% rename(eg15R1=rawse)
eg21R1<-subset(rep1,generation=="G21") %>% select(gene,rawse) %>% rename(eg21R1=rawse)
eg30R1<-subset(rep1,generation=="G30") %>% select(gene,rawse) %>% rename(eg30R1=rawse)
eg40R1<-subset(rep1,generation=="G40") %>% select(gene,rawse) %>% rename(eg40R1=rawse)


rep2<-subset(d,replicate=="R2")
eg6R2<-subset(rep2,generation=="G6") %>% select(gene,rawse) %>% rename(eg6R2=rawse)
eg15R2<-subset(rep2,generation=="G15") %>% select(gene,rawse) %>% rename(eg15R2=rawse)
eg21R2<-subset(rep2,generation=="G21") %>% select(gene,rawse) %>% rename(eg21R2=rawse)
eg30R2<-subset(rep2,generation=="G30") %>% select(gene,rawse) %>% rename(eg30R2=rawse)
eg40R2<-subset(rep2,generation=="G40") %>% select(gene,rawse) %>% rename(eg40R2=rawse)

rep3<-subset(d,replicate=="R3")
eg6R3<-subset(rep3,generation=="G6") %>% select(gene,rawse) %>% rename(eg6R3=rawse)
eg15R3<-subset(rep3,generation=="G15") %>% select(gene,rawse) %>% rename(eg15R3=rawse)
eg21R3<-subset(rep3,generation=="G21") %>% select(gene,rawse) %>% rename(eg21R3=rawse)
eg30R3<-subset(rep3,generation=="G30") %>% select(gene,rawse) %>% rename(eg30R3=rawse)
eg40R3<-subset(rep3,generation=="G40") %>% select(gene,rawse) %>% rename(eg40R3=rawse)

lm1<-edgeme(nR1,nR2,nR3,eg30R1,eg40R1)
lm2<-edgeme(nR1,nR2,nR3,eg30R2,eg40R2)
lm3<-edgeme(nR1,nR2,nR3,eg30R3,eg40R3)
dflm1<-data.frame(gene=rownames(lm1$table),logfc=lm1$table$logFC, pval=lm1$table$PValue, fdr=p.adjust(lm1$table$PValue,method="BH"))
dflm2<-data.frame(gene=rownames(lm2$table),logfc=lm2$table$logFC, pval=lm2$table$PValue, fdr=p.adjust(lm2$table$PValue,method="BH"))
dflm3<-data.frame(gene=rownames(lm3$table),logfc=lm3$table$logFC, pval=lm3$table$PValue, fdr=p.adjust(lm3$table$PValue,method="BH"))

lmvs2<-edgemeext(eg30R1,eg40R1,eg30R2,eg40R2,eg30R3,eg40R3)
dflmvs2<-data.frame(gene=rownames(lmvs2$table),logfc=lmvs2$table$logFC, pval=lmvs2$table$PValue,fdr=p.adjust(as.numeric(lmvs2$table$PValue),method="BH"))

#### Coloring FDR 
fdr<-0.01
fdr2<-0.001

dflm1$col<-0
dflm1[dflm1$fdr<fdr,]$col<-1
dflm1[dflm1$fdr<fdr2,]$col<-2
dflm1$col<-as.factor(dflm1$col)

dflm2$col<-0
dflm2[dflm2$fdr<fdr,]$col<-1
dflm2[dflm2$fdr<fdr2,]$col<-2
dflm2$col<-as.factor(dflm2$col)

dflm3$col<-0
dflm3[dflm3$fdr<fdr,]$col<-1
dflm3[dflm3$fdr<fdr2,]$col<-2
dflm3$col<-as.factor(dflm3$col)

dflmvs2$col<-0
dflmvs2[dflmvs2$fdr<fdr,]$col<-1
#dflmvs2[dflmvs2$fdr<fdr2,]$col<-2
dflmvs2$col<-as.factor(dflmvs2$col)

pvs2<-ggplot(data = dflmvs2, aes(y = -log10(pval), x = logfc,color=col )) +
  geom_point(alpha = 1, size = 2)+scale_colour_manual(values=palete)+theme(legend.position = "none")+
  ylab("-log10(p)")+xlab("log2(fold-change)")

p1<-ggplot(data = dflm1, aes(y = -log10(pval), x = logfc,color=col )) +
  geom_point(alpha = 1, size = 2)+scale_colour_manual(values=palete)+theme(legend.position = "none")+
  ylab("-log10(p)")+xlab("log2(fold-change)")

p2<-ggplot(data = dflm2, aes(y = -log10(pval), x = logfc,color=col )) +
  geom_point(alpha = 1, size = 2)+scale_colour_manual(values=palete)+theme(legend.position = "none")+
  ylab("-log10(p)")+xlab("log2(fold-change)")

p3<-ggplot(data = dflm3, aes(y = -log10(pval), x = logfc,color=col )) +
  geom_point(alpha = 1, size = 2)+scale_colour_manual(values=palete)+theme(legend.position = "none")+
  ylab("-log10(p)")+xlab("log2(fold-change)")

 
#pdf(file="",width=8,height=3)

#grid.arrange(p1, p2, p4, ncol=3)
#dev.off()

#pdf(file="",width=3,height=3)
plot(pvs2)
#dev.off()

```



