---
title: "DNA analysis"
author: "Matthew Beaumont"
date: "`r Sys.Date()`"
output: github_document
---

```{bash, eval=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Read trimming

All of the following .fq.gz read files were trimmed to the minimum read length of the shortest reads from all files (100) for uniformity.

```{bash}
cd /Volumes/Data/Projects/DmelR2_P-ele/fastq/raw 
ls -lh
```

```{bash, eval=FALSE}
nohup bash scripts/trim.sh > trim.log &
```

```{bash, eval=FALSE}
for i in fastq/*.fq.gz
        do 
                date
                j=`basename $i`
                j=${j%.fq.gz}
                zless $i | awk '{print substr($1,1,100)}' | gzip -9 > fastq/trimmed/${j}_trimmed.fq.gz
                date
        done
```

The following trimmed .fq.gz files were generated.

```{bash}
cd /Volumes/Data/Projects/DmelR2_P-ele/fastq/trimmed
ls -lh
```

# TE Mapping

The fast-miner.sh script was altered to accommodate the altered file name for the trimmed read files.

```{bash, eval=FALSE}
nohup zsh fastq-miner-trimmed.sh mel /Volumes/Data/Projects/DmelR2_P-ele/fastq/trimmed > /Volumes/Data/Projects/DmelR2_P-ele/logs/mel.log &
```

```{bash, eval=FALSE}
if [ $# -lt 2 ]
  # "$#" is number of parameters- here we test
   then
   echo "Usage $0 sampleid inputdir"
exit 2
fi

set -o shwordsplit

source ~/.zshrc

# Files and folders
sampleid=$1
inputdir=$2
outputdir=$2
outabu="../results/$sampleid/rpm"
outbamall="../results/$sampleid/bam"
tmpdir="../tmp"

## the REFERENCE 
refg="../refg/Dmel_tes_3scg/teseqs-3scg.fasta"
hier="../refg/Dmel_tes_3scg/teseqs.hier"

# software
samtools="../bin/samtools"
bwa="../bin/bwa"
popte2="../bin/popte2.jar"
readtorpm="../bin/readstat2rpm_all.py"

mkdir -p $outabu
mkdir -p $outbamall
# paths
for read1 in $inputdir/*_1_trimmed.fq.gz
do
	# variables defined to store the sra ids of the files from the input directory
        
	read2=${read1%_1_trimmed.fq.gz}
	read2=${read2}_2_trimmed.fq.gz
	tmp=`basename $read1`
        readid=${tmp%_1.fq.gz}

	tmpfile=$tmpdir/${readid}.fq.gz
	
	gzip -cd $read1 $read2 | paste - - - - |awk '{print "@" NR,$2,"+" NR,$4}'|tr " " "\n" |gzip -c > $tmpfile
	#gzip -cd $read1 $read2 |gzip -c > $tmpfile
 
        # mapping
	bamfile="$tmpdir/$readid.sort.bam"
	command="$bwa bwasw -t 8 -M $refg $tmpfile | $samtools view -Sb - | $samtools sort -T $sraid.nnnn.bam -O bam -m 4G > $bamfile"
	echo "executing mapping $command"
	eval $command
 
	# TE bam
	allbamfile="$outbamall/$readid.allte.sort.bam"
	samtools view -b -F 0x004 $bamfile > $allbamfile
	
	# PopoolationTE2
	opteabu="$outabu/$readid.rawabu"
	opterpm="$outabu/$readid.rpm"
	echo "Writting raw abundance to $opteabu"
	java -jar $popte2 stat-reads --bam $bamfile --map-qual 10 --hier $hier --output $opteabu
	echo "Writting rpm to $opterpm"
	python $readtorpm --rs $opteabu > $opterpm 

	# Cleanup tmp
        rm $bamfile
	rm $tmpfile
done 
```

This altered fast-miner.sh TEMiner script was then used to generate the following .bam and .rpm files from the trimmed .fq.gz read files.

```{bash}
cd /Volumes/Data/Projects/DmelR2_P-ele/software/teminer-code-r10/results/mel/bam
ls -lh

cd /Volumes/Data/Projects/DmelR2_P-ele/software/teminer-code-r10/results/mel/rpm
ls -lh
```

# P-element copy number estimates

The deviate-family.sh script was run on the resulting .bam and .rpm files, to assess and visualise P-element presence, using the family ID "PPI251".

```{bash, eval=FALSE}
nohup zsh deviate-family.sh mel PPI251 > /Volumes/Data/Projects/DmelR2_P-ele/logs/deviate_mel &
```

```{bash, eval=FALSE}

if [ $# -lt 2 ]
  # "$#" is number of parameters- here we test
   then
   echo "Usage $0 sampleid family"
exit 2
fi

set -o shwordsplit

# Files and folders
sampleid=$1
tefamily=$2
output="../results/$sampleid/deviate/$tefamily"
inputbam="../results/$sampleid/bam"
# scgold="Dmel_rpl32,Dmel_piwi,Dmel_Act5C"
scg="Dmel_rhi,Dmel_rpl32,Dmel_tj"

## the REFERENCE 
refg="../refg/Dmel_tes_3scg/teseqs-3scg.fasta"
anno="../refg/Dmel_tes_3scg/teseqs.gff"
hier="../refg/Dmel_tes_3scg/teseqs.hier"


# software
samtools="../bin/samtools"
bwa="../bin/bwa"

mkdir -p $output

# paths
for bam in $inputbam/*.allte.sort.bam
do
	
	n=`basename $bam` 
	sampleid=${n%.allte.sort.bam}
	echo $sampleid


	com="deviate --input_bam $bam --library $refg --annotation $anno --single_copy_genes $scg --families $tefamily --minID 1"
	echo $com
	eval $com
	mv ${bam}.${tefamily} $output/$sampleid.$tefamily
	mv ${bam}.${tefamily}.pdf  $output/${sampleid}.${tefamily}.pdf
	mv ${bam}.${tefamily}.raw $output/${sampleid}.${tefamily}.raw
	rm ${bam}.fused.sort.bam
	rm ${bam}.fused.sort.bam.bai

	
done 
```

Generating the following .fq.gz.PPI251 files.

```{bash}
cd /Volumes/Data/Projects/DmelR2_P-ele/software/teminer-code-r10/results/mel/deviate/PPI251
ls -lh *fq.gz.*
```

From the resulting .fq.gz.PPI251 files, we extracted and assembled the P-element copy number values into a single file.

```{bash, eval=FALSE}

for i in *.PPI251; do echo $i | cut -f1 -d "." | cut -f2 -d "_"  >> mel_pcopies.txt; grep "or " $i | cut -f5 -d " "  | cut -b 1-8 >> mel_pcopies.txt; done

 mel_pcopies.txt |grep  '^R' -A2 | xargs -n3 > mel_pcopies_aligned.txt

done
```

```{bash}
cd /Volumes/Data/Projects/DmelR2_P-ele/software/teminer-code-r10/results/mel/deviate/PPI251
ls -lh *pcopies*
```

# Visualisation

## P-element copy number

The copy number data was then visualised using ggplot2.

```{r}
library (ggplot2)
theme_set(theme_bw())
tresrep<-c("firebrick", "skyblue3", "chartreuse4")
  
d = read.table("mel_pcopies_values.txt")

names(d)<-c ("replicate", "generation", "copy number")

print(d)

ggplot(d, aes(x = generation, y = `copy number`, group= replicate)) + 
  geom_line(aes(color=replicate),linewidth=1) +
  geom_point(aes(color=replicate)) +
  ggtitle("P-element invasions in D. melanogaster") +
  scale_colour_manual(values=tresrep) +
  xlim(0, 65)

ggsave("figs/pele_invasions_dmel.png", width = 8, height = 5, dpi = 600)

knitr::include_graphics("figs/pele_invasions_dmel.png")

```

## DeviaTE

We also generated DeviaTE plots for each replicate for each measured generation time-point.

```{R}
knitr::include_graphics("figs/pele_coverage.png")

```

END.
